src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -314,6 +321,177 @@ export default class GroupComparisonStore extends ComparisonStore {
+                        s.uniqueSampleKey,
+                        p.molecularProfileId,
+                        this.selectedMutationMapperGene!.hugoGeneSymbol,
+                        this.coverageInformation.result!
+                    )
+                )
+            ).length;
+        },
+    });
+
+    readonly allSamples = remoteData({
+        await: () => [this._session],
+        invoke: async () => {
+            const allStudies = _(this._session.result!.groups)
+                .flatMapDeep(groupData => groupData.studies.map(s => s.id))
+                .uniq()
+                .value();
+            // fetch all samples - faster backend processing time
+            const allSamples = await client.fetchSamplesUsingPOST({
+                sampleFilter: {
+                    sampleListIds: allStudies.map(studyId => `${studyId}_all`),
+                } as SampleFilter,
+                projection: 'DETAILED',
+            });
+
+            return allSamples;
+        },
+    });
+
+    readonly genePanelDataForMutationProfiles = remoteData({
+        await: () => [this.allSamples, this.mutationEnrichmentProfiles],
+        invoke: async () => {
+            const sampleMolecularIdentifiers = getSampleMolecularIdentifiers(
+                this.allSamples.result!,
+                this.mutationEnrichmentProfiles.result!
+            );
+            const genePanelData = client.fetchGenePanelDataInMultipleMolecularProfilesUsingPOST(
+                {
+                    genePanelDataMultipleStudyFilter: {
+                        sampleMolecularIdentifiers,
+                    } as GenePanelDataMultipleStudyFilter,
+                }
+            );
+            return genePanelData;
+        },
+    });
+
+    readonly coverageInformation = remoteData<CoverageInformation | undefined>({
+        await: () => [
+            this.genePanelDataForMutationProfiles,
+            this.sampleKeyToSample,
+            this.patients,
+        ],
+        invoke: () => {
+            return Promise.resolve(
+                getCoverageInformation(
+                    this.genePanelDataForMutationProfiles.result!,
+                    this.sampleKeyToSample.result!,
+                    this.patients.result!,
+                    [this.selectedMutationMapperGene!]
+                )
+            );
+        },
+    });
+
+    readonly patients = remoteData({
+        await: () => [this.allSamples],
+        invoke: () => fetchPatients(this.allSamples.result!),
+        default: [],
+    });
+
+    readonly availableGenes = remoteData<Gene[]>({
+        invoke: async () => {
+            const genes = await getAllGenes();
+            return genes.sort((a, b) =>
+                a.hugoGeneSymbol < b.hugoGeneSymbol ? -1 : 1
+            );
+        },
+    });
+
+    public setDefaultGene() {
+        this._selectedMutationMapperGene = this.genesWithMaxFrequency[0].hugoGeneSymbol;
+    }
+
+    @computed get selectedMutationMapperGene() {
```

Comment:
activeMutationMapperGene so that conceptually it accomodatds both the default and user selected state

Reply:
then you can have another computed "userSelectedMutationMapperGene" which returns the raw _userSelectedMutationMapperGene